components/esp_insights/src/esp_insights.c

```diff
@@ -94,7 +94,7 @@ typedef struct {
     int data_msg_id;
     uint32_t data_msg_len;
     SemaphoreHandle_t data_lock;
-    char app_sha256[APP_ELF_SHA256_LEN];
+    uint8_t app_sha256[32];
```

Comment:
Please do not use magic numbners.
Also, as we discussed, array of 32 shouldn't be needed.